var/spack/repos/builtin/packages/hipcub/package.py

```diff
@@ -133,20 +151,32 @@ class Hipcub(CMakePackage):
         "5.4.0",
         "5.4.3",
     ]:
-        depends_on("hip@" + ver, when="@" + ver)
-        depends_on("rocprim@" + ver, when="@" + ver)
+        depends_on("rocprim@" + ver, when="+rocm @" + ver)
         depends_on("rocm-cmake@%s:" % ver, type="build", when="@" + ver)
 
+    # fix hardcoded search in /opt/rocm and broken config mode search
+    patch("find-hip-cuda-rocm-5.1.patch", when="@5.1:5.2 +cuda")
+    patch("find-hip-cuda-rocm-5.3.patch", when="@5.3: +cuda")
+
     def setup_build_environment(self, env):
```

Comment:
i see setup_build_environment had been removed for other recipes in the PR except for hipcub. Any reason why. Otherwise i am okay with the changes.

Reply:
hipblas, hipsolver, hipfft and hipsparse are all written in C++ and can be built with a normal C++ compiler, but hipCUB is written in the HIP language and requires a HIP compiler.